### PR TITLE
Set better default commands for loading images - take 2

### PR DESCRIPTION
**Pull request by dims**
Signed-off-by: Davanum Srinivas <davanum@gmail.com>

**What type of PR is this?**
/kind bug

**What this PR does / why we need it**:
Do not default to docker, check the runtime and use correct command for each runtime.

**Which issue(s) this PR fixes**:

Fixes #

**Special notes for your reviewer**:
I made a mistake! https://github.com/kubernetes/kubernetes/commit/01183e51f084b1a05420f743bca3011b3925a084#diff-6a44976362555568e0243a480da2e86b need to re-apply what was in https://github.com/kubernetes/kubernetes/pull/92184

**Does this PR introduce a user-facing change?**:

```release-note
NONE
```

**Additional documentation e.g., KEPs (Kubernetes Enhancement Proposals), usage docs, etc.**:

```docs

```

### CODE REVIEW

**Comment by BenTheElder** [comment]
[borrowing this PR to test something non-blocking, will `/skip` it after verifying]
/test  pull-kubernetes-e2e-kind-canary

File: cluster/gce/gci/configure.sh [hunk 1]
```diff
@@ -376,9 +376,18 @@ function try-load-docker-image {
...
   local -r max_attempts=5
   local -i attempt_num=1
-  until timeout 30 ${LOAD_IMAGE_COMMAND:-docker load -i} "${img}"; do
+
+  if [[ "${CONTAINER_RUNTIME_NAME:-}" == "docker" ]]; then
+    load_image_command=${LOAD_IMAGE_COMMAND:-docker load -i}
+  elif [[ "${CONTAINER_RUNTIME_NAME:-}" == "containerd" ]]; then
+    load_image_command=${LOAD_IMAGE_COMMAND:-ctr -n=k8s.io images import}
+  else
+    load_image_command="${LOAD_IMAGE_COMMAND:-}"
```
Comment (BenTheElder) [inline cluster/gce/gci/configure.sh:385]: I think we shouldn't allow this else to succeed if the command is empty? maybe:
```suggestion
    load_image_command="${LOAD_IMAGE_COMMAND:?}"
```
or better yet an explicit check + error message

Comment (BenTheElder) [reply cluster/gce/gci/configure.sh:385]: the latter is probably best. `:?` will just ensure it was set.

Comment (bsdnet) [reply cluster/gce/gci/configure.sh:385]: If it is unsupported runtime, it should fail at line 390. 
In theory, failing earlier will be better. But I did not see we
have functions like assert or exit in this file.

**Comment by bsdnet** [comment]
From the latest failures, looks environmental variable load_image_command is not set.

https://storage.googleapis.com/kubernetes-jenkins/logs/ci-containerd-e2e-gci-gce/1275221697632407556/artifacts/bootstrap-e2e-master/kube-master-installation.log

So that means CONTAINER_RUNTIME_NAME is not set properly when calling configure.sh

**Comment by bsdnet** [comment]
I am wondering why ${CONTAINER_RUNTIME_NAME:-}, why not ${CONTAINER_RUNTIME_NAME} if CONTAINER_RUNTIME_NAME is one environmental varaible.

**Comment by bsdnet** [comment]
I have a new PR #92401, but still tried to piece things together :)

**Pull request merged**